### ISSUE TITLE
About labels

### ISSUE THREAD

**Issue by Broccoli77PP** [OMEGA-RS/S3Net_CD#2]
Thank you for improving your source code, but I would like to ask why the labeled image data is added to the model training in the code, since the labeled image data is not used in the paper

**Comment by OMEGA-RS** [comment]
Thanks for your question. In fact, only the generated labels from the pretext task are used to train our model in the phase of model building, aiming to obtain the pretrained model to learn useful features of the input data. In the phase of label predict, the label image is used to evaluate the accuracy of the proposed method.

**Comment by Broccoli77PP** [comment]
<img width="469" alt="屏幕截图 2024-03-05 222512" src="https://github.com/OMEGA-RS/S3Net_CD/assets/135418964/1cc5fcc1-bca8-404e-b3fa-05f5b6406064">
Thank you for your prompt response. Can I understand that REF_PATH in My Dataset here does not participate in the final change detection process.

**Comment by OMEGA-RS** [comment]
Yes, here, the ref_path is only used to load the real label (i.e., the reference image) for evaluating the performance of the proposed method. In the model prediction phase, the real label does not participate.

**Comment by Broccoli77PP** [comment]
> 是的，在这里，ref_path仅用于加载真实标签（即参考图像）以评估所提出的方法的性能。在模型预测阶段，实标签不参与。
Thank you very much for your reply

**Comment by Broccoli77PP** [comment]
Thank you very much for your reply

**Comment by cdzzprog** [comment]
你好，我问一下这个Seg_Label.mat这个数据是咋弄的呀

**Comment by cdzzprog** [comment]
你好，我问一下这个对主成分分析后的数据进行超像素分割后是保存为mat格式嘛，这两个参数该怎么调整n_segments， compactness（n_segments 是分割的超像素数目，compactness 控制超像素的形状）

**Comment by cdzzprog** [comment]
你好，我问一下这个超像素分割的算法是基于matlab的嘛

**Comment by OMEGA-RS** [comment]
同学你好，超像素分割是采用VL-FEAT工具包中的vl_slic算法（matlab）实现的，一般默认分割尺度为10，rate设为0.2，针对不同数据集rate的值可能会有所改变，OpenCV中的slic算法效果稍微差一点，可自行调整相关适参数，希望能够对你有所帮助


2025-3-20









---- Replied Message ----
From ***@***.***> Date 3/20/2025 10:08 To ***@***.***> Cc Tao ***@***.***>,
***@***.***> Subject Re: [OMEGA-RS/S3Net_CD] About labels (Issue #2)  
你好，我问一下这个超像素分割的算法是基于matlab的嘛
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***>
cdzzprog left a comment (OMEGA-RS/S3Net_CD#2)
你好，我问一下这个超像素分割的算法是基于matlab的嘛

**Comment by cdzzprog** [comment]
好的，感谢你的回答

**Comment by cdzzprog** [comment]
还有个问题想请教一下，这种算法能和域适应结合嘛
